daemon/oci_windows.go

```diff
@@ -288,6 +288,28 @@ func (daemon *Daemon) createSpecWindowsFields(c *container.Container, s *specs.S
 				if cs, err = readCredentialSpecRegistry(c.ID, csValue); err != nil {
 					return err
 				}
+			} else if match, csValue = getCredentialSpec("config://", splitsOpt[1]); match {
+				// if the container does not have a DependencyStore, then it
+				// isn't swarmkit managed. In order to avoid creating any
+				// impression that `config://` is a valid API, return the same
+				// error as if you'd passed any other random word.
+				if c.DependencyStore == nil {
+					return fmt.Errorf("invalid credential spec security option - value must be prefixed file:// or registry:// followed by a value")
```

Comment:
`errdefs.InvalidParameter(errors.Errorf(...))`

Reply:
I can do that, but this function is littered with `return fmt.Errorf` calls. It is called by `createSpec`, which is in turn called in `(*Daemon).containerStart`, where the error result is wrapped (in `daemon/start.go`, on line 154, with `errdefs.System`:

https://github.com/moby/moby/blob/740349757396d8f1ad573d4b78148baca9c979aa/daemon/start.go#L154-L157

Reply:
🤦‍♂️ That error conversion is incorrect.